### PR TITLE
Add Privacy Policy and Terms of Service Page

### LINKED ISSUE #8353: Feature: Add Privacy Policy and Terms of Service Page
## Description

Gitea, just like GitHub, GitLab, and Bitbucket, is essentially a service to users.

Currently Gitea doesn't have Privacy Policy and Terms of Service. Privacy Policy lists how Gitea (Gitea instance) collect and manage user's private data. Terms of Service provide legal binding between Gitea and users which must be agreed in order to use Gitea.

Regarding full legal code, we can provide `starter` template which can be edited by site administrators. They can either using online generators or hiring a lawyer to write the legal code based on the template.

This feature request can be helpful especially in EU when GDPR is in effective.

### PR DESCRIPTION
**Pull request by bagasme**
Both legal pages are implemented in static HTML, located in `public/privacy.html` and `public/tos.html`, respectively.

Currently uses lorem ipsum texts, which will be replaced by real legal code / clauses. Please add more commits here to add them (legal code).

This will close #8353.

### CODE REVIEW

**Comment by lafriks** [comment]
I don't think no one here has legal background to update these texts, I would prefer that everything would be moved to contrib (static sample file) and add section in docs to describe how to add this. Translations can be left tho

**Comment by zeripath** [comment]
I think it would be better to start from this: https://gdpr.eu/wp-content/uploads/2019/01/Our-Company-Privacy-Policy.pdf instead of lorem ipsum dolorum.

In reality the vast majority of gitea installs will follow the same pattern so, if we can get the boilerplate right for a default install then most people should never need to change it.

**Review by techknowlogick — DISMISSED** [review]
Blocking merge per @lafriks feedback

**Comment by bagasme** [comment]
@lafriks So basically you proposed like [https://docs.gitea.io/en-us/customizing-gitea/#adding-links-and-tabs](adding extra links)?

**Comment by bagasme** [comment]
Thanks for suggestion @lafriks

**Comment by bagasme** [comment]
@techknowlogick @lafriks done

File: docs/content/doc/advanced/adding-legal-pages.en-us.md
Comment (sapk) [inline docs/content/doc/advanced/adding-legal-pages.en-us.md]: Even if I like the term `hack`, I would suggest the more common word `edit`.
I would suggest also to use the term `need` since it there is some engagement in it that are not managed by Gitea like:.
```
we will delete your full profile (within reason) within 90 days of your request
``` 
```suggestion
Now you need to edit the page to meet your requirements.
```

**Comment by bagasme** [comment]
For milestone, I expect this PR to be merged for 1.11 or 1.12.

**Comment by techknowlogick** [comment]
@bagasme as this is a docs PR that doesn't apply to a specific version it can be merged as soon as it is approved without milestone.

**Comment by bagasme** [comment]
@techknowlogick so can this PR be merged soon?

**Comment by techknowlogick** [comment]
@bagasme yes, once two LG-TM given, and no pending requested changes.

File: docs/content/doc/advanced/customizing-gitea.en-us.md
Comment (sapk) [inline docs/content/doc/advanced/customizing-gitea.en-us.md]: links in others parts of docs seems to not be relative.
```suggestion
For more information, see [Adding Legal Pages](https://docs.gitea.io/en-us/adding-legal-pages).
```

**Review by sapk — CHANGES_REQUESTED** [review]
I have only quickly read the legal notice and it seems ok. 
The docs part indicating that it need to be review and the fact that they are in contrib/ folder is good enough. 
Just need one little fix for relative link otherwise L-GTM.

File: docs/content/doc/advanced/customizing-gitea.en-us.md
Comment (bagasme) [reply docs/content/doc/advanced/customizing-gitea.en-us.md]: @sapk Hmm... I think you misunderstood relative and absolute link. Your suggestion actually use absolute link (there is `docs.gitea.io` domain in the URL), while I use relative link (from docs root).

Comment (bagasme) [reply docs/content/doc/advanced/customizing-gitea.en-us.md]: BTW, I will apply it.

File: contrib/legal/privacy.html.sample
Comment (zeripath) [inline contrib/legal/privacy.html.sample]: `<i>` is an inline element not a block level element.

Use a `<h3>` or `<h4>` etc.

```suggestion
  <h4>Last updated: December 28, 2019</h4>
```

File: contrib/legal/privacy.html.sample
Comment (zeripath) [inline contrib/legal/privacy.html.sample]: If we're gonna close list elements we should close them all.

```suggestion
   <li>Profile information for your Account (such as your full name, biography, website, and location.)</li>
```

File: contrib/legal/privacy.html.sample
Comment (zeripath) [inline contrib/legal/privacy.html.sample]:
```suggestion
  <p>However, we don't intentionally collect sensitive information (such as racial or ethnic origin, political affiliations, religious/philosophical beliefs, biometric data, etc.) If you choose to store any of such data on our servers, you are responsible for complying with any regulations regarding them.</p>
```

File: contrib/legal/privacy.html.sample
Comment (bagasme) [reply contrib/legal/privacy.html.sample]: Oh, I forgot to close this `<li>` listing!

File: contrib/legal/privacy.html.sample
Comment (zeripath) [inline contrib/legal/privacy.html.sample]:
```suggestion
   <p>We share your User Personal Information, if you consent, after letting you know what information will be shared, with whom, and why. For example, if you allow third party applications to access your Account using <a href="https://docs.gitea.io/en-us/oauth2-provider/">OAuth2 providers</a>, we share all information associated with your Account, including private repos and organizations. You may also direct us through your action on Your Gitea Instance to share your User Personal Information, such as when joining an Organization.</p>
```

File: contrib/legal/privacy.html.sample
Comment (zeripath) [inline contrib/legal/privacy.html.sample]: This is probably overkill and I doubt most instances will be doing it. It's fine to keep it in but it may wrongly give the impression that you can't run a Gitea legally without doing this. 

You probably couldn't store very sensitive personal information without doing this and so it makes sense to advertise if your Gitea is suitable for that but I doubt many services can say they control their own racks and cages.

File: contrib/legal/privacy.html.sample
Comment (zeripath) [inline contrib/legal/privacy.html.sample]:
```suggestion
  <p><b>Disclaimer:</b> No method of transmission, or method of electronic storage, is 100% secure, therefore, we cannot guarantee absolute security.</p>
```

File: contrib/legal/privacy.html.sample
Comment (zeripath) [inline contrib/legal/privacy.html.sample]: Gitea out of the box doesn't do this. I guess it's reasonable to prepare a tos and privacy statement to cover reasonable extensions but we should be careful not to put off our privacy caring users.

File: contrib/legal/privacy.html.sample
Comment (bagasme) [reply contrib/legal/privacy.html.sample]: @zeripath Data breach mitigation overkill for small and medium deployments?

**Review by sapk — APPROVED** [review]

File: contrib/legal/privacy.html.sample
Comment (zeripath) [reply contrib/legal/privacy.html.sample]: Running a Gitea instance on aws or vps is excluded by these terms - it's not just basic data breach mitigation.

This level of protection is appropriate for medical data. It may not be suitable for a fork of Kodi protected by a 10 character password. Similarly encrypting the storage of publicly available source code which is verifiable with signatures may not be necessary - particularly as the most likely source of a leak is not going to be a data store attack but rather a security fault in Gitea, or the stack running Gitea - in which case that the filesystem is encrypted is of no consequence.

A service should advertise the level of security they are engaging in. At what point you are negligent under the GDPR has not been formally tested but I doubt it extends to owning your own server racks and cages.

It's fine to keep it in as ideal but we should remember that most users aren't running Gitea as a GitHub competitor. The average Gitea user runs this a private host for themselves plus/minus a few others. They still need to comply with the GDPR and we want to make it easy for them to do so.

We need sensible defaults.

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]: <i> is not a block level element - use h4

```suggestion
  <h4>Last updated: December 31, 2019</h4>
```

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]: This appears to be a trailing close p tag.

```suggestion
```

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]: Trailing close p
```suggestion
```

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]:
```suggestion
   <p>We only offer support via email, in-Service communications, and electronic messages. We do not offer telephone support.</p>
```

File: docs/content/doc/advanced/adding-legal-pages.en-us.md
Comment (zeripath) [inline docs/content/doc/advanced/adding-legal-pages.en-us.md]:
```suggestion
Now you need to edit the page to meet your requirements. In particular you must change the email addresses, web addresses and references to "Your Gitea Instance" to match your situation.

You absolutely must not place a general ToS or privacy statement that implies that the gitea project is responsible for your server.
```

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]: ol is a block level element

```suggestion
```

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]: ol is a block level element

```suggestion
```

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]: ol is a block level element

```suggestion
```

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]: ol is a block level element

```suggestion
```

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]: ol is a block level element

```suggestion
```

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]: ol is a block level element

```suggestion
```

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]: We do not want emails to copyright@gitea.io from everyone's instances

```suggestion
  <p>If you are copyright owner and believe that content on our website violates your copyright, please contact us at <a href="mailto:copyright@your-gitea-instance">copyright@your-gitea-instance</a>. Please note that before sending a takedown notice, consider legal uses (such as fair use and licensed use); and legal consequences for sending false notices.</p>
```

File: contrib/legal/privacy.html.sample
Comment (zeripath) [inline contrib/legal/privacy.html.sample]: Privacy@gitea.io will not be a helpful email address for most instances

```suggestion
  <p>If you have any concerns about privacy, please contact us at <a href="mailto:privacy@your-gitea-instance">privacy@your-gitea-instance</a>. We will respond promptly, within 45 days.</p>
```

File: contrib/legal/tos.html.sample
Comment (zeripath) [inline contrib/legal/tos.html.sample]:
```suggestion
    <li>The "Website" refers to Your Gitea Instance's website at <a href="https://your-gitea-instance">your-gitea-instance</a>, including its subdomains and other websites owned by Your Gitea Instance.</li>
```

**Comment by zeripath** [comment]
Thanks @bagasme docs PR s can get very annoying with lots of nit picking hopefully my suggestions are useful and can mostly be batched up to be committed as one.

I do think we might need some graduated levels or more explicit advice on what needs to be in there (esp regarding third party data sharing.) But those could be dealt with as other documentation PRs.

Thanks once again.

File: contrib/legal/privacy.html.sample
Comment (bagasme) [reply contrib/legal/privacy.html.sample]: @zeripath so what sensible defaults then?

File: contrib/legal/tos.html.sample
Comment (bagasme) [reply contrib/legal/tos.html.sample]: @zeripath I think the corresponding `<p>` opening tag right after account terms `<h2>` should also be deleted, because it is the pair of `</p>` you mentioned.

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
@go-gitea/owners I think this one could do with a final sign-off.

This is a good start, however, we will likely need to make some clarifications in the docs pages - as it could be interpreted that all gitea instances use third-party resources and/or that the only way to have a compliant Gitea is to host it on its own multisite server racks with full disk encryption etc.

**Comment by sapk** [comment]
I think the next step would be to build a file generator outside gitea (or a plugin 😄) depending on question (instance name, resonse time, third party, ...) that the user would deploy on their instance.

File: docs/content/doc/advanced/adding-legal-pages.en-us.md
Comment (lunny) [inline docs/content/doc/advanced/adding-legal-pages.en-us.md]: /path/to/custom/public/privacy.html

Comment (bagasme) [reply docs/content/doc/advanced/adding-legal-pages.en-us.md]: OK

File: docs/content/doc/advanced/adding-legal-pages.en-us.md
Comment (6543) [inline docs/content/doc/advanced/adding-legal-pages.en-us.md]: i would replace this with the online link

Comment (bagasme) [reply docs/content/doc/advanced/adding-legal-pages.en-us.md]: @6543 did you mean replacing this `cp` line with following?:
```suggestion
wget -O /path/to/custom/public/privacy.html https://raw.githubusercontent.com/go-gitea/gitea/master/contrib/legal/privacy.html.sample
```

File: docs/content/doc/advanced/adding-legal-pages.en-us.md
Comment (6543) [inline docs/content/doc/advanced/adding-legal-pages.en-us.md]:
```suggestion
Create or append to `/path/to/custom/templates/custom/extra_links_footer.tmpl`:
```

Comment (6543) [reply docs/content/doc/advanced/adding-legal-pages.en-us.md]: it normaly is on the footer ...

File: contrib/legal/privacy.html.sample
Comment (6543) [inline contrib/legal/privacy.html.sample]:
```suggestion
  <p>However, we don't intentionally collect sensitive information (such as racial or ethnic origin, political affiliations, religious/philosophical beliefs, biometric data, etc.)</p>
<!--If you choose to store any of such data on our servers, you are responsible for complying with any regulations regarding them.-->
```

File: contrib/legal/privacy.html.sample
Comment (6543) [inline contrib/legal/privacy.html.sample]:
```suggestion
   <li>Profile information for your Account (such as your full name, biography, website, gpg key, and location.)</li>
```

File: contrib/legal/privacy.html.sample
Comment (6543) [inline contrib/legal/privacy.html.sample:26]:
```suggestion
   <li>Git data that you upload to a repository</li>
   <li>Cookies and Similar Technologies</li>
```

**Review by 6543 — CHANGES_REQUESTED** [review]
smal nit

**Review by 6543 — APPROVED** [review]
.

**Comment by 6543** [comment]
there is law support for open source but i dont have the connections ...
 * https://fsfe.org/activities/ftf/ln.de.html

maby somebody can wirte someone on mastodnon ...

**Pull request merged**
